pkg/upgraders/osd/upgrader.go

```diff
@@ -126,9 +126,14 @@ func PreClusterHealthCheck(c client.Client, cfg *osdUpgradeConfig, scaler scaler
 	if err != nil {
 		return false, err
 	}
+
 	desired := upgradeConfig.Spec.Desired
 	if upgradeCommenced {
-		logger.Info(fmt.Sprintf("ClusterVersion is already set to Channel %s Version %s, skipping %s", desired.Channel, desired.Version, upgradev1alpha1.UpgradePreHealthCheck))
+		if len(desired.Channel) > 0 && len(desired.Version) > 0 {
+			logger.Info(fmt.Sprintf("ClusterVersion is already set to Channel %s Version %s, skipping %s", desired.Channel, desired.Version, upgradev1alpha1.UpgradePreHealthCheck))
+		} else if len(desired.Image) > 0 && len(desired.Version) > 0 {
+			logger.Info(fmt.Sprintf("ClusterVersion is already set to Image %s, skipping %s", desired.Image, upgradev1alpha1.UpgradePreHealthCheck))
+		}
 		return true, nil
```

Comment:
we should either fail here if the version is not provided or make it required in the type definition

Reply:
This part is removed from upgrade and moved to clusterversion package..